File: java/Problem1021.java
import java.util.Scanner;

public class Problem1021 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        double value = sc.nextDouble();

        int cents = (int) (value * 100);

        System.out.println("NOTAS:");

        System.out.printf("%d nota(s) de R$ 100.00\n", (cents / 10000));

        cents %= 10000;

        System.out.printf("%d nota(s) de R$ 50.00\n", (cents / 5000));

        cents %= 5000;

        System.out.printf("%d nota(s) de R$ 20.00\n", (cents / 2000));

        cents %= 2000;

        System.out.printf("%d nota(s) de R$ 10.00\n", (cents / 1000));

        cents %= 1000;

        System.out.printf("%d nota(s) de R$ 5.00\n", (cents / 500));

        cents %= 500;

        System.out.printf("%d nota(s) de R$ 2.00\n", (cents / 200));

        cents %= 200;

        System.out.println("MOEDAS:");


        System.out.printf("%d moeda(s) de R$ 1.00\n", (cents / 100));

        cents %= 100;

        System.out.printf("%d moeda(s) de R$ 0.50\n", (cents / 50));

        cents %= 50;

        System.out.printf("%d moeda(s) de R$ 0.25\n", (cents / 25));

        cents %= 25;

        System.out.printf("%d moeda(s) de R$ 0.10\n", (cents / 10));

        cents %= 10;

        System.out.printf("%d moeda(s) de R$ 0.05\n", (cents / 5));

        cents %= 5;

        System.out.printf("%d moeda(s) de R$ 0.01\n", cents );

        sc.close();
    }
}

File: python/problem1021.py
valor = float(input())
x = int(round(valor * 100))  # Converte o valor para centavos e arredonda para evitar imprecisão

oneHundred = x // 10000
x = x % 10000

fifty = x // 5000
x = x % 5000

twenty = x // 2000
x = x % 2000

ten = x // 1000
x = x % 1000

five = x // 500
x = x % 500

two = x // 200
x = x % 200

coinOne = x // 100
x = x % 100

coinFifty = x // 50
x = x % 50

coinTF = x // 25
x = x % 25

coinTen = x // 10
x = x % 10

coinFive = x // 5
x = x % 5

coin = x  # O restante já está em centavos

print('NOTAS:')
print(f"{oneHundred} nota(s) de R$ 100.00")
print(f"{fifty} nota(s) de R$ 50.00")
print(f"{twenty} nota(s) de R$ 20.00")
print(f"{ten} nota(s) de R$ 10.00")
print(f"{five} nota(s) de R$ 5.00")
print(f"{two} nota(s) de R$ 2.00")

print('MOEDAS:')
print(f"{coinOne} moeda(s) de R$ 1.00")
print(f"{coinFifty} moeda(s) de R$ 0.50")
print(f"{coinTF} moeda(s) de R$ 0.25")
print(f"{coinTen} moeda(s) de R$ 0.10")
print(f"{coinFive} moeda(s) de R$ 0.05")
print(f"{coin} moeda(s) de R$ 0.01")

